src/io/data_buffers/distributed_io_buffer.cpp

```diff
@@ -181,6 +181,10 @@ void lbann::distributed_io_buffer::calculate_num_iterations_per_epoch(int num_mo
     max_mini_batch_size = data_reader->get_num_data();
   }
```

Comment:
We should check somewhere here and in the partitioned_io_buffer that the max_mini_batch_size is properly computed for each model and that it is not aggregate (sum) for all models. The global/max_mini_batch_size should be as specified by the user in the prototext/cmd line.